README.md

```diff
@@ -198,11 +207,82 @@ A numeric value that can be parsed by [`(float)` cast, equivalent to `floatval`]
 ###### `bool` default
 `true`, `on`, `yes` or `1` will result in true. Other values, as long as there is something, will result default false. (If there is nothing, the variable will not be optional)
 
+#### Example of using variables
+##### SQL file
+```sql
+-- #{ example
+-- #    { insert
+-- # 	  :foo string
+-- # 	  :bar int
+INSERT INTO example(
+	foo_column
+	bar_column
+) VALUES (
+	:foo
+	:bar
+);
+-- #    }
+-- #    { select
+-- # 	  :foo string
+-- # 	  :bar int
+SELECT * FROM example
+WHERE foo_column = :foo
+LIMIT :bar;
+-- #    }
+-- #}
+```
+##### Code
+```php
+// Example of using variable in insert statements
+$this->database->excuteInsert("example.insert", ["foo" => "sample text", "bar" => 123]);
+
+// Example of using variable in select statements
+$this->database->executeSelect("example.select", ["foo" => "sample text", "bar" => 1]);
+```
+
 ### Query text
 Query text is not a command, but the non-commented part between the start and end commands of a query declaration.
 
 Variables are interpolated in query text using the `:var` format. Note that libasynql uses a homebrew algorithm for identifying the variable positions, so they might be inaccurate.
+```sql
+-- #{ query.declarartion
+SELECT * FROM example;
+-- The line above is a query text
+-- #}
+```
+
+## Things to beware
+### Common mistakes
+```php
+public $foo = 'bar';
+
+public function setFoo() : void {
+	$this->foo = 'foo';
+}
+
+public function getFoo() : string {
+	return $this->foo;
+}
+```
+
+```php
+$this->database->executeGeneric("common.mistake.asynchronous", [], function() : void {
+	$this->setFoo();
+});
+echo $this->getFoo();
+```
+The result will be `bar` because the queries are ran in asynchronous, the code in the main thread will always <!-- Correct me if I'm wrong -->run faster than it.
```

Comment:
```suggestion
The result will be `bar` because the queries are run asynchronously. The code on the main thread will run faster than it.
```

Reply:
Huh "asynchronously" is a thing? Google translate shouted at me when I type it.